packages/connectivity/src/scp-cf/destination/destination-service.ts

```diff
@@ -186,6 +235,11 @@ async function fetchDestinationByTokens(
 
   return callDestinationService(targetUri, authHeader, options)
     .then(response => {
+      if (isCertificateResponse(response.data)) {
+        throw new Error(
+          'Destination service response must be destination response.'
+        );
+      }
```

Comment:
I am just wondering: Why is this check necessary? Is it possible to get such a response here?
I didn't find the test for this either.

Reply:
Oh I guess this is just a type guard. Still, can this happen?

Reply:
no, perhaps you have a better idea. we have the callDestinationService which calls the service for a given uri. The return type could be a destination or certificate. Perhaps it is better to have two separate functions than the typeguard I guess.

Reply:
I think you already did what you described in the last sentence 😊 👍